pymodbus/client/mixin.py

```diff
@@ -546,16 +546,16 @@ def convert_from_registers(
 
     @classmethod
     def convert_to_registers(
-        cls, value: Union[int, float, str], data_type: DATATYPE
+        cls, value: Union[float, str], data_type: DATATYPE
```

Comment:
it is important for the user to know he can call with an "int" and do not need to convert to float.

e.g.
```
  value = 127.0, data_type = DATASTYPE.INT16
```
is not legal but:
```
  value = 127, data_type = DATASTYPE.INT16
  value = 127.2, data_type = DATASTYPE.FLOAT32
```
are both legal.